cms/teams/management/commands/sync_teams.py

```diff
@@ -0,0 +1,135 @@
+        return groups
+
+    @transaction.atomic
+    def update_teams(self, groups: Iterable[dict], *, dry_run: bool) -> None:
+        """Updates the local database with the teams fetched from the API."""
+        # Fetch existing Teams from the database
+        existing_teams: dict[str, Team] = {team.identifier: team for team in Team.objects.all()}
+        active_team_ids = {team.identifier for team in existing_teams.values() if team.is_active}
+        api_team_groups = [group for group in groups if group["id"] not in settings.ROLE_GROUP_IDS]
+        teams_from_api: set[str] = {group["id"] for group in api_team_groups}
+
+        for group in api_team_groups:
+            self._update_team(group, existing_teams=existing_teams, dry_run=dry_run)
+
+        # Deactivate teams not present in the API
+        teams_to_deactivate = set(active_team_ids) - teams_from_api
+        if teams_to_deactivate:
+            self.stdout.write(f"Deactivating {len(teams_to_deactivate)} team(s) not present in the API.")
+            for team_id in teams_to_deactivate:
+                self.stdout.write(f"Deactivating team: {team_id} - {existing_teams[team_id].name}")
+
+            if not dry_run:
+                Team.objects.filter(identifier__in=teams_to_deactivate).update(is_active=False)
+
+    def _update_team(self, group: Mapping, *, existing_teams: Mapping[str, Team], dry_run: bool) -> None:
+        """Updates a single team in the database."""
+        group_id: str = group["id"]
+        name: str = group["name"].strip()
+        precedence: int = group["precedence"]
+        created_str: str = group["creation_date"]
+        updated_str: str = group["last_modified_date"]
+
+        created_at: datetime = datetime.fromisoformat(created_str).replace(tzinfo=UTC)
+        updated_at: datetime = datetime.fromisoformat(updated_str).replace(tzinfo=UTC)
+
+        if group_id in existing_teams:
+            team = existing_teams[group_id]
+            if (
+                team.name != name
+                or team.precedence != precedence
+                or team.created_at != created_at
+                or team.updated_at != updated_at
+            ):
+                self.stdout.write(f"Updating team: {group_id}")
+                if not dry_run:
+                    team.name = name
+                    team.precedence = precedence
+                    team.created_at = created_at
+                    team.updated_at = updated_at
+                    team.is_active = True
+                    team.save()
+        else:
+            self.stdout.write(f"Creating new team: {name} (ID: {group_id})")
+            if not dry_run:
+                Team.objects.create(
+                    identifier=group_id,
+                    name=name,
+                    precedence=precedence,
+                    created_at=created_at,
+                    updated_at=updated_at,
+                    is_active=True,
+                )
```

Comment:
Suggestion: I think this is just re-implementing `create_or_update`. It might be cleaner to reuse that method.